### PR TITLE
CI: mpi-pytest follow-up deselections

### PR DESCRIPTION
**Pull request by ethanglaser**
### Description

Follow-up to https://github.com/intel/scikit-learn-intelex/pull/1777. Deselecting some sporadically failing tests while resolution is underway.

_Add a comprehensive description of proposed changes_

_List issue number(s) if exist(s): #6 (for example)_

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py [hunk 1]
```diff
@@ -151,8 +151,9 @@ def test_logistic_spmd_synthetic(n_samples, n_features, C, tol, dataframe, queue
     ).fit(dpt_X_train, dpt_y_train)
 
     # TODO: Logistic Regression coefficients do not align
-    tol = 1e-2
-    assert_allclose(spmd_model.coef_, batch_model.coef_, rtol=tol, atol=tol)
+    tol = 1e-1
+    # TODO: Re-enable coefficient check when oneDAL fix implemented
+    # assert_allclose(spmd_model.coef_, batch_model.coef_, rtol=tol, atol=tol)
```
Comment (samir-nasibli) [inline sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:156]: is it possible to design it via pytest.skip?

Comment (ethanglaser) [reply sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:156]: It would require a full redesign since multiple checks occur in each function and I am not skipping entire function

**Review by md-shafiul-alam — APPROVED** [review]
Since this is a temporary skip, and solution is in progress, I guess we can merge this.

**Review by samir-nasibli — DISMISSED** [review]
Please address comment below

File: sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py (hunk 1 above)
Comment (ethanglaser) [reply sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:156]: I disagree since we are still getting some meaningful validation, especially in functions where both fit() and predict() are being checked - I don't think it would make sense to skip entire pipeline just because one small part is not passing

Comment (samir-nasibli) [reply sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:156]: My point is that the worst thing to disable test is commenting the code. We've had several refactoring steps where we tried to avoid this, let's  follow this.

Now it turns out that the tests are passed, and in the logs it will be that the tests passed, which is a little confusing for me if not the whole case passed.

@icfaust @Alexsandruss I'd be glad to hear your opinions.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Test should be explicitly skipped rather than commented.

File: sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py [hunk 2]
```diff
@@ -116,6 +116,8 @@ def test_logistic_spmd_gold(dataframe, queue):
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 @pytest.mark.mpi
 def test_logistic_spmd_synthetic(n_samples, n_features, C, tol, dataframe, queue, dtype):
+    # TODO: Re-enable coefficient check when oneDAL fix implemented
+    pytest.skip("Test disabled while OneDAL fix in progress")
```
Comment (samir-nasibli) [inline sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:120]: I think it is better cover it with ticket and avoid TODO commenting here.
```suggestion
    pytest.skip("Sporadic failures on coefficient check. Test disabled while OneDAL fix in progress")
```

File: sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py [hunk 3]
```diff
@@ -116,6 +116,8 @@ def test_logistic_spmd_gold(dataframe, queue):
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 @pytest.mark.mpi
 def test_logistic_spmd_synthetic(n_samples, n_features, C, tol, dataframe, queue, dtype):
+    # TODO: Re-enable coefficient check when oneDAL fix implemented
```
Comment (ethanglaser) [inline sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:119]:
```suggestion
```

**Review by samir-nasibli — APPROVED** [review]
Thank you! Looks good to me! Assuming green CI

File: sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py [hunk 4]
```diff
@@ -116,6 +116,7 @@ def test_logistic_spmd_gold(dataframe, queue):
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 @pytest.mark.mpi
 def test_logistic_spmd_synthetic(n_samples, n_features, C, tol, dataframe, queue, dtype):
+    pytest.skip("Sporadic failures on coefficient check. Test disabled while OneDAL fix in progress")
```
Comment (ethanglaser) [inline sklearnex/spmd/linear_model/tests/test_logistic_regression_spmd.py:119]:
```suggestion
    pytest.skip("Sporadic failures on coef_ check. Test disabled while fix in progress")
```

**Comment by ethanglaser** [comment]
http://intel-ci.intel.com/ef54d025-ee4e-f1e4-9f5d-a4bf010d0e2e

**Review by Alexsandruss — APPROVED** [review]
